src/inmanta/data/__init__.py

```diff
@@ -4519,6 +4519,34 @@ async def get_resources_in_latest_version(
                         result.append(resource)
         return result
 
+    @classmethod
+    async def get_latest_resources_resource_type_count(cls, environment: uuid.UUID) -> JsonType:
+        """
+        Returns the count for each resource_type over all resources in the model's latest version
+        """
+        query_latest_model = f"""
+            SELECT max(version)
+            FROM {ConfigurationModel.table_name()}
+            WHERE environment=$1
+        """
+        query = f"""
+            SELECT resource_type, count(*) as count
+            FROM (
+                SELECT DISTINCT resource_id, resource_type
+                FROM {Resource.table_name()}
+                WHERE environment=$1
+                AND model=({query_latest_model})
+            ) AS r1
+            GROUP BY resource_type;
```

Comment:
This query is more complex than it needs to be. It's important to realize that resource_ids are unique within a given model version of a given environment. So there is no need for the DISTINCT here.

```suggestion
            SELECT resource_type, count(*) as count
            FROM {Resource.table_name()}
            WHERE environment=$1 AND model=({query_latest_model})
            GROUP BY resource_type;
```